wallet/src/de/schildbach/wallet/ui/WalletAddressesFragment.java

```diff
@@ -187,6 +189,27 @@ public boolean onActionItemClicked(final ActionMode mode, final MenuItem item) {
 
                     mode.finish();
                     return true;
+
+                case R.id.wallet_addresses_private_key:
+                    // Not in clipboard for security reasons, unless on testnet.
+                    final String privKey = getKey(position).getPrivateKeyAsWiF(Constants.NETWORK_PARAMETERS);
+                    if (Constants.TEST) {
+                        clipboardManager.setPrimaryClip(ClipData.newPlainText("Bitcoin privKey", privKey));
+                        log.info("private key copied to clipboard: {}", privKey);
+                    }
+                    final DialogBuilder builder = DialogBuilder.warn(activity, R.string.action_private_key);
+                    builder.setMessage(privKey);
+                    builder.singleDismissButton(new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialogInterface, int i) {
+                            mode.finish();
```

Comment:
I only close the ActionMode there so that in foreground you can read the privKey and in the background you can read the pubKey as the Toolbar's title while the Dialog is shown.